lib/puppet/provider/rvm_system_ruby/rvm_system_ruby.rb

```diff
@@ -21,7 +21,7 @@ def destroy
   def exists?
     begin
       rvmcmd("list", "strings").split("\n").any? do |line|
-        line =~ Regexp.new(Regexp.escape(resource[:name]))
+        line =~ Regexp.new(/^resource[:name]$/)
```

Comment:
If you're going to match the whole line, why not simplify it to
```suggestion
        line == resource[:name]
```

And if you go that far, you can reduce it to
```ruby
rvmcmd("list", "strings").split("\n").include?(resouce[:name])

Reply:
Submitted as https://github.com/voxpupuli/puppet-rvm/pull/183